File: core/FindPeakElement/FindPeakElement.py
from typing import List


def findPeakElement(nums: List[int]) -> int:
    length = len(nums)
    left, right = 0, length - 1
    while left <= right:
        mid = (left + right) // 2
        if (mid == 0 or nums[mid] > nums[mid - 1]) and (mid == length - 1 or nums[mid] > nums[mid + 1]):
            return mid
        if (mid == 0 and nums[mid] < nums[mid + 1]) or nums[mid] < nums[mid + 1]:
            left = mid + 1
        else:
            right = mid - 1

File: core/FindPeakElement/FindPeakElement.java
package com.dbc;

public class FindPeakElement {
    public int findPeakElement(int[] nums) {
        int left = 0, right = nums.length - 1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if ((mid == 0 || nums[mid] > nums[mid - 1]) && (mid == nums.length - 1 || nums[mid] > nums[mid + 1])) return mid;
            if (nums[mid] < nums[mid + 1]) left = mid + 1;
            else right = mid - 1;
        }
        return 0;
    }
}
